internal/olm/operator/registry/fbcindex/fbc_registry_pod.go

```diff
@@ -233,8 +237,11 @@ func (f *FBCRegistryPod) podForBundleRegistry(cs *v1alpha1.CatalogSource) (*core
 		return nil, fmt.Errorf("configMap error: %w", err)
 	}
 
-	volumes := []corev1.Volume{}
-	volumeMounts := []corev1.VolumeMount{}
+	var (
+		volumes            []corev1.Volume
+		sharedVolumeMounts []corev1.VolumeMount
+		gzipVolumeMount    []corev1.VolumeMount
```

Comment:
Is there a specific reason we need to have a separate variable for this? Can we just add the gzip volume mount to the sharedVolumeMounts if the gzip flag is enabled?

Reply:
For the regular usecasem these volumeMountes point to the configMaps, while in the gzip usecase, they point to the uncompressed volumes.

The `gzipVolumeMount` slice only used in the gzip usecase, and its volumeMounts point to the configMaps. 

So in the gzip usecase, we want `sharedVolumeMountslice` slice both in t init container and in the registry container, and we want the `gzipVolumeMount` only in the init container.